oauthenticator/oauth2.py

```diff
@@ -70,13 +70,16 @@ def _OAUTH_ACCESS_TOKEN_URL(self):
     def _OAUTH_USERINFO_URL(self):
         return self.authenticator.userdata_url
 
-    def set_state_cookie(self, state):
-        self._set_cookie(STATE_COOKIE_NAME, state, expires_days=1, httponly=True)
+    def set_state_cookie(self, state_cookie_value):
+        self._set_cookie(
+            STATE_COOKIE_NAME, state_cookie_value, expires_days=1, httponly=True
+        )
 
-    _state = None
+    def _generate_state_id(self):
+        return uuid.uuid4().hex
 
-    def get_state(self):
-        next_url = original_next_url = self.get_argument("next", None)
+    def _get_next_url(self):
+        next_url = self.get_argument("next", None)
```

Comment:
`get_state` was there to allow subclasses to override state generation (e.g. to use another opaque string, or it could remove the next_url from the state). I don't think we need really need this, nor has anyone used it. It was never documented.

Part of the reason I don't think it's something to be concerned about losing is that the serialization of state is assumed in `check_state` (could also be overridden, I suppose), and the serialization APIs are themselves private.

I've been thinking of a way to make a deprecated `get_state()` that can be overridden, but that's complicated by the fact that there's two states now - the cookie and the URL parameter, when the true 'state' is really just the state ID.

So I think it's okay to lose this as an undocumented API in 16.1.

Reply:
Wieee excellent, thank you @minrk for making an informed decision about this!!